### PR TITLE
GROOVY-10024: Dollar slashy /$ string ending in backslash "breaks" th…

### PR DESCRIPTION
**Pull request by paulk-asert**
…e parser

### CODE REVIEW

**Comment by danielsun1106** [comment]
LGTM. Let's wait for the test result:
https://github.com/danielsun1106/groovy-parser/tree/GROOVY-10024

**Comment by danielsun1106** [comment]
The test result is green ;-)

File: src/test/groovy/StringTest.groovy [hunk 1]
```diff
@@ -241,6 +241,11 @@ foo
 
         text = $/$$$$$//$
         assert text == '$$/'
+
+        //GROOVY-10024
+        def s1 = $/abc\/$
+        def s2 = $/def/$
+        assert s1.size() == 4 && s2.size() == 3
```
Comment (pbielicki) [inline src/test/groovy/StringTest.groovy:248]: I’d assert the expected values instead of the size.

Comment (timyates) [reply src/test/groovy/StringTest.groovy:248]: Does

```
def s1 = $/Failing string\\/$
```

With double backslash at the end work (as per the Jira example)?

Comment (paulk-asert) [reply src/test/groovy/StringTest.groovy]: The double backslash case works - I'll adjust the test accordingly and also with expected values.

Comment (timyates) [reply src/test/groovy/StringTest.groovy]: Awesome as always Paul 😎👍

**Comment by paulk-asert** [comment]
Proposed PR merged.

**Pull request closed without merging**